x-pack/plugins/security_solution/server/usage/detections_helpers.ts

```diff
@@ -0,0 +1,148 @@
+}
+
+const isElasticRule = (tags: string[]) => tags.includes(`${INTERNAL_IMMUTABLE_KEY}:true`);
+
+const initialRuleUsage: DetectionRulesUsage = {
+  detection_rules_custom_enabled: 0,
+  detection_rules_custom_disabled: 0,
+  detection_rules_elastic_enabled: 0,
+  detection_rules_elastic_disabled: 0,
+};
+
+const initialMlJobUsage: MlJobsUsage = {
+  ml_jobs_custom_enabled: 0,
+  ml_jobs_custom_disabled: 0,
+  ml_jobs_elastic_enabled: 0,
+  ml_jobs_elastic_disabled: 0,
+};
+
+export const buildRuleUsage = (rulesMetrics: DetectionsMetric[]): DetectionRulesUsage =>
+  rulesMetrics.reduce((stats, { isEnabled, isElastic }) => {
+    if (isEnabled && isElastic) {
+      return {
+        ...stats,
+        detection_rules_elastic_enabled: stats.detection_rules_elastic_enabled + 1,
+      };
+    } else if (!isEnabled && isElastic) {
+      return {
+        ...stats,
+        detection_rules_elastic_disabled: stats.detection_rules_elastic_disabled + 1,
+      };
+    } else if (isEnabled && !isElastic) {
+      return {
+        ...stats,
+        detection_rules_custom_enabled: stats.detection_rules_custom_enabled + 1,
+      };
+    } else if (!isEnabled && !isElastic) {
+      return {
+        ...stats,
+        detection_rules_custom_disabled: stats.detection_rules_custom_disabled + 1,
+      };
+    } else {
+      return stats;
+    }
+  }, initialRuleUsage);
+
+export const buildMlJobUsage = (jobMetrics: DetectionsMetric[]): MlJobsUsage =>
+  jobMetrics.reduce((stats, { isEnabled, isElastic }) => {
+    if (isEnabled && isElastic) {
+      return {
+        ...stats,
+        ml_jobs_elastic_enabled: stats.ml_jobs_elastic_enabled + 1,
+      };
+    } else if (!isEnabled && isElastic) {
+      return {
+        ...stats,
+        ml_jobs_elastic_disabled: stats.ml_jobs_elastic_disabled + 1,
+      };
+    } else if (isEnabled && !isElastic) {
+      return {
+        ...stats,
+        ml_jobs_custom_enabled: stats.ml_jobs_custom_enabled + 1,
+      };
+    } else if (!isEnabled && !isElastic) {
+      return {
+        ...stats,
+        ml_jobs_custom_disabled: stats.ml_jobs_custom_disabled + 1,
+      };
+    } else {
+      return stats;
+    }
+  }, initialMlJobUsage);
+
+export const fetchRules = async (
+  index: string,
+  callCluster: LegacyAPICaller
+): Promise<DetectionsMetric[]> => {
+  let ruleMetrics: DetectionsMetric[] = [];
+  const ruleSearchOptions: SearchParams = {
+    body: { query: { bool: { filter: { term: { 'alert.alertTypeId': SIGNALS_ID } } } } },
+    filterPath: ['hits.hits._source.alert.enabled', 'hits.hits._source.alert.tags'],
+    ignoreUnavailable: true,
+    index,
+    size: 10000, // elasticsearch index.max_result_window default value
+  };
+  try {
+    const ruleResults = await callCluster<{ alert: { enabled: boolean; tags: string[] } }>(
+      'search',
+      ruleSearchOptions
+    );
+
+    if (ruleResults.hits?.hits?.length > 0) {
+      ruleMetrics = ruleResults.hits.hits.map((hit) => ({
```

Comment:
Not really an issue, but why not do the tally here rather than iterating twice?

Reply:
That's a good point, I'll change these to do a single loop 👍